### ISSUE TITLE
object-literal-sort-keys and accented words

### ISSUE THREAD

**Issue by rodrigojcmello** [palantir/tslint#4167]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.0.3
- __Running TSLint via__:  VSCode

#### TypeScript code being linted

```ts
interface IAdicionarItem {
    autor: string; // these 3 items were sorted alphabetically by vscode (no tslint was used for this), here it is right.
    título: string; // i with accent 
    type: 'ADICIONAR_ITEM';
}

export const adicionarItem = (autor: string, título: string): IAdicionarItem => ({
    autor,
    título, // "i" with accent should still come before "y"
    type: 'ADICIONAR_ITEM', // [tslint] The key 'type' is not sorted alphabetically (object-literal-sort-keys)
});
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "warning",
    "extends": [
        "tslint:recommended",
        "tslint-react"
    ],
    "jsRules": {},
    "rules": {
        "quotemark": [
            true,
            "single",
            "jsx-single"
        ],
        "jsx-no-multiline-js": false,
        "jsx-no-lambda": false,
        "no-console": false
    },
    "rulesDirectory": []
}
```

#### Actual behavior
Words with accents must be in the end so there is no error / warning in TSLint

#### Expected behavior
Items should be sorted alphabetically considering accents without any TSLint error / warning

<img width="1280" alt="captura de tela 2018-09-16 as 11 18 32" src="https://user-images.githubusercontent.com/5174526/45597590-fae6b980-b9a4-11e8-8877-3834a84e232c.png">

**Comment by pe8ter** [comment]
Looks like JavaScript has some options for an easy win:

https://davidwalsh.name/sorting-strings-accented-characters
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

I could make a PR if this. I'd like to know if using `localeCompare()` for all sorting would be an acceptable solution, or if adding an option to use this as the sort function would be preferable.

**Comment by johnwiseheart** [comment]
I think its probably preferable to add an option for this to avoid throwing errors on code that was previously "correct". In the next major version we can turn on the option by default.

**Comment by pe8ter** [comment]
@rodrigojcmello Feature released in https://github.com/palantir/tslint/releases/tag/5.12.0.

**Comment by johnwiseheart** [comment]
Since it’s been released, I think we can close this issue.
